In [1]:
import numpy as np

In [2]:
import matplotlib.pyplot as plt

In [3]:
import seaborn as sns

In [4]:
 import pandas as pd  

In [5]:
gdp_df=pd.read_csv('../data/gdp_percapita.csv')

In [6]:
gdp_df

Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2021  1517.016266              NaN
1        Afghanistan  2020  1968.341002              NaN
2        Afghanistan  2019  2079.921861              NaN
3        Afghanistan  2018  2060.698973              NaN
4        Afghanistan  2017  2096.093111              NaN
...              ...   ...          ...              ...
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

[7662 rows x 4 columns]

In [7]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB


How many rows and columns are in `gdp_df`? 7662 rows x 4 columns
What are the data types of each column? object, int64, float64

In [8]:
gdp_df=gdp_df.drop(gdp_df.columns[[3]], axis=1)

In [14]:
gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2021     1517.016266
1     Afghanistan  2020     1968.341002
2     Afghanistan  2019     2079.921861
3     Afghanistan  2018     2060.698973
4     Afghanistan  2017     2096.093111
...           ...   ...             ...
7657     Zimbabwe  1994     2670.106615
7658     Zimbabwe  1993     2458.783255
7659     Zimbabwe  1992     2468.278257
7660     Zimbabwe  1991     2781.787843
7661     Zimbabwe  1990     2704.757299

[7662 rows x 3 columns]

In [16]:
gdp_df=gdp_df.rename(columns={"Country or Area": "Country", "Value": "GDP_Per_Capita"})

In [13]:
gdp_df[gdp_df["Country"]=="Afghanistan"]

Country  Year  GDP_Per_Capita
0   Afghanistan  2021     1517.016266
1   Afghanistan  2020     1968.341002
2   Afghanistan  2019     2079.921861
3   Afghanistan  2018     2060.698973
4   Afghanistan  2017     2096.093111
5   Afghanistan  2016     2101.422187
6   Afghanistan  2015     2108.714173
7   Afghanistan  2014     2144.449634
8   Afghanistan  2013     2165.340915
9   Afghanistan  2012     2122.830759
10  Afghanistan  2011     1961.096317
11  Afghanistan  2010     2026.163818
12  Afghanistan  2009     1823.742614
13  Afghanistan  2008     1556.844452
14  Afghanistan  2007     1528.344578
15  Afghanistan  2006     1366.993146
16  Afghanistan  2005     1352.320627
17  Afghanistan  2004     1260.060581
18  Afghanistan  2003     1292.333437
19  Afghanistan  2002     1280.463171

In [26]:
gdp_df["Year"]. nunique()

33

In [30]:
country_year_count=gdp_df["Country"].value_counts().reset_index()

In [31]:
country_year_count.head()

Country  count
0  Least developed countries: UN classification     33
1                    Middle East & North Africa     33
2       Middle East & North Africa (IDA & IBRD)     33
3                                 Middle income     33
4                                      Mongolia     33

In [35]:
len(country_year_count[country_year_count["count"]==33])

202

How many countries have data for all years? 202

In [36]:
len(country_year_count[country_year_count["count"]<20])

7

Which countries are missing many years of data? 7 have less than 20 years of data

In [39]:
country_year_count["count"].value_counts()

count
33    202
28     10
32      5
23      4
31      3
22      3
29      2
26      2
10      2
30      1
25      1
24      1
20      1
19      1
16      1
15      1
14      1
12      1
Name: count, dtype: int64

In [ ]:
unique_country=gdp_df.drop_duplicates(subset='Country or Area')

In [ ]:
print(unique_country)

In [ ]:
plt.hist(gdp_df.GDP_Per_Capita)

In [ ]:
plt.show()

In [ ]:
plt.gaussian_kde(gdp_df.value)

In [ ]:
plt.boxplot(gdp_df.value, gdp_df.year)

In [ ]:
plt.show()

In [ ]:
mean_gdp=gdp_df["value"].mean()

In [ ]:
print(mean_gdp)